### PR TITLE
Optimizes the reverse walker

### PR DESCRIPTION
**Pull request by zonble**
Also

- Use tab key to fast iterate the candidates
- Add UI to set composing buffer size

### CODE REVIEW

File: Source/KeyHandler.mm [hunk 1]
```diff
@@ -1276,11 +1358,11 @@ - (void)_walk
 
     // if DEBUG is defined, a GraphViz file is written to kGraphVizOutputfile
 #if DEBUG
-    std::string dotDump = _builder->grid().dumpDOT();
-    NSString *dotStr = [NSString stringWithUTF8String:dotDump.c_str()];
-    NSError *error = nil;
-
-    BOOL __unused success = [dotStr writeToFile:kGraphVizOutputfile atomically:YES encoding:NSUTF8StringEncoding error:&error];
+//    std::string dotDump = _builder->grid().dumpDOT();
```
Comment (lukhnos) [inline Source/KeyHandler.mm]: Should this be uncommented, or should the entire `#if DEBUG` section be deleted?

File: Source/KeyHandler.mm [hunk 2]
```diff
@@ -573,6 +578,83 @@ - (BOOL)handleInput:(KeyHandlerInput *)input state:(InputState *)inState stateCa
...
+{
+    if (![state isKindOfClass:[InputStateInputting class]]) {
+        errorCallback();
+        return YES;
+    }
+    if (!_bpmfReadingBuffer->isEmpty()) {
+        errorCallback();
+        return YES;
+    }
+    // zonble
```
Comment (lukhnos) [inline Source/KeyHandler.mm]: Extra comment?

**Review by lukhnos — APPROVED** [review]
LGTM

File: Source/KeyHandler.mm (hunk 2 above)
Comment (zonble) [reply Source/KeyHandler.mm]: OK. Let me fix it.

**Pull request merged**